### ISSUE TITLE
Consider longer function names for certain functions

### ISSUE THREAD

**Issue by mbylstra** [lynn/elm-arithmetic#1]
First up - this package looks really useful! I've used Native to change number bases in the past - a pure Elm function is much better.

Is there a reason cbrt and gcd aren't named cubeRoot and greatestCommonDivisor? Seems out of place with the other functions that are in plain English. Perhaps these names are common in MATLAB or numpy, but full English is far more approachable for math noobs like myself.

**Comment by lynn** [comment]
Hello, thanks for the feedback! Elm itself calls its square root function `Basics.sqrt`, so I wanted to name my own function in a similar fashion — that's why it's a bit of an oddball. However, I prefer descriptive names, too. I think I might just export my own `squareRoot` and `cubeRoot` functions.

For GCD and LCM, the abbreviations are so common that I feared the long names would be unwieldy. I'll think about it — maybe I could export both? How descriptive their names _need_ to be depends on the problem domain, so it's a bit tricky.

**Comment by lynn** [comment]
Note to self: `numDivisors` is weirdly abbreviated, too. I'll make it `divisorCount` when I get home.

**Comment by mbylstra** [comment]
Ah yes - good point about Basics.sqrt. `sqrt` is such a common function in all programming languages that I wouldn't have even noticed that it's an abbreviation - it's pretty much part of programming vernacular. I've never ventured far outside the standard libraries for my maths needs, so I'd never seen the abbreviations gcd and lcm before. It makes sense to keep them abbreviated if they are often used functions and well known by mathematicians. Yes, it's a tricky balance - it does depend on how often something is used. Eg: `ls` wouldn't make any sense to someone that's never used a command line, but it would extremely annoying to have to type something like `list_files` every time. Having a function exported with more than one name doesn't seem very common in libraries - I guess the "it's best to have one way of doing things" rule of thumb applies.

**Comment by lynn** [comment]
I renamed the functions, and added a couple more :)

**Comment by mbylstra** [comment]
Awesome! It will be really interesting to see elm-arithmetic grow - what could be better than a pure functional language for doing math.
